### PR TITLE
chore: Bring latest changes from master into dev branch (includes adopting latest stable SDK version)

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): CLOUDP-266281

Brings changes from master which include using latest SDK version 2024-08-05. The `CLOUDP-212932-dev` branch no longer relies on the the SDK dev preview after merging this PR.

**Note**: For reviewing would recommend to focus on documentation, examples, and implementation of resources that have been modified in this dev branch: `advanced_cluster`, `cloud_backup_schedule`, and `cluster` (only docs). Specific merge conflicts related to these resources were made after the initial merge commit [f045058](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2491/commits/f0450581d7ca3250fd8058234b65f36e75dad4ab).

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Comment by AgustinBettati** [comment]
Tests in `stream` test group are failing due to pending Atlas fix (CLOUDP-265822)

**Review by EspenAlbert — APPROVED** [review]
LGTM. Although a bit challenging.
I'm comparing with master to get fewer changes.
Checking my understanding:
1. After this the merge with master we can directly merge `dev` to master since all conflicts have been resolved?
2. Does this branch now contain the extra admin client that is locked to `connV220240530` to ensure resources behave the same?
3. All extra changes to advanced_cluster (ISS features and docs) will be in the diff when merging to master?

**Review by maastha — APPROVED** [review]

**Comment by AgustinBettati** [comment]
@EspenAlbert yes understand from reviewer perspective it can be more complex, left all manual conflict resolutions in separate commits to make evident separation from trivial changes that came from master.

1. After this the merge with master we can directly merge dev to master since all conflicts have been resolved?

Correct.

2. Does this branch now contain the extra admin client that is locked to connV220240530 to ensure resources behave the same?

Yes, connV220240530 is now being used to consume previous API versions in `advanced_cluster` (for specific scenarios), `cloud_backup_schedule` (for specific scenarios), and `global_cluster_config` (fully). ISS dev branch was already doing this but with another SDK version (connV220231115).

3. All extra changes to advanced_cluster (ISS features and docs) will be in the diff when merging to master?

Yes, once this is merged the diff of dev branch with master will be ISS (advanced_cluster and cloud_backup_schedule) and cluster deprecation (docs changes).

File: internal/config/client.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ const (
 type MongoDBClient struct {
 	Atlas           *matlasClient.Client
 	AtlasV2         *admin.APIClient
-	AtlasV220231115 *admin20231115.APIClient
+	AtlasV220240530 *admin20240530.APIClient // used in advanced_cluster and cloud_backup_schedule for avoiding breaking changes
```
Comment (marcosuma) [inline internal/config/client.go:34]: just double checking. This is correct right? @AgustinBettati

**Review by marcosuma — APPROVED** [review]

File: internal/config/client.go (hunk 1 above)
Comment (AgustinBettati) [reply internal/config/client.go:34]: Yes, with 2024-08-05 released (and used for AtlasV2 client), 2024-05-30 is the nearest SDK version that still consumes the older versions.

**Pull request merged**